services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -486,13 +499,20 @@ async def _get_link_for_directory_fmd(
             raise S3KeyNotFoundError(key=file_id, bucket=self.simcore_bucket_name)
         return await self.__get_link(parse_obj_as(SimcoreS3FileID, file_id), link_type)
 
-    async def delete_file(self, user_id: UserID, file_id: StorageFileID):
+    async def delete_file(
+        self,
+        user_id: UserID,
+        file_id: StorageFileID,
+        *,
+        enforce_access_rights: bool = True,
+    ):
         async with self.engine.acquire() as conn, conn.begin():
-            can: AccessRights | None = await get_file_access_rights(
-                conn, user_id, file_id
-            )
-            if not can.delete:
-                raise FileAccessRightError(access_right="delete", file_id=file_id)
+            if enforce_access_rights:
+                can: AccessRights | None = await get_file_access_rights(
+                    conn, user_id, file_id
+                )
+                if not can.delete:
```

Comment:
here is a bug. If `can=None` then `can.delete` will raise an exception. What does `None` mean in this context?